autopilot/contractor.go

```diff
@@ -694,37 +718,40 @@ func (c *contractor) runContractFormations(ctx context.Context, w Worker, hosts
 		}
 	}
 
+	// create a gouging checker
+	gc := worker.NewGougingChecker(state.gs, state.rs, cs, state.fee, state.cfg.Contracts.Period, state.cfg.Contracts.RenewWindow)
+
 	// calculate min/max contract funds
 	minInitialContractFunds, maxInitialContractFunds := initialContractFundingMinMax(state.cfg)
 
+	var stalled bool
 	for h := 0; missing > 0 && h < len(candidates); h++ {
-		if c.ap.isStopped() {
-			break
-		}
-
 		host := candidates[h]
 
 		// break if the autopilot is stopped
 		if c.ap.isStopped() {
 			break
 		}
 
-		// fetch a new price table if necessary
-		if err := refreshPriceTable(ctx, w, &host); err != nil {
-			c.logger.Errorf("failed to fetch price table for candidate host %v: %v", host.PublicKey, err)
-			continue
+		// if we were stalled refreshing the price table, we fetch the consensus
+		// state and recreate the gouging checker to ensure we're checking host
+		// height against an updated renter height
+		if stalled {
```

Comment:
Here I would do something like have a `lastStateUpdate` and check whether the last update was more than 60 seconds or so ago. Considering that we are looking at candidates for forming contracts here chances are that we don't have a valid price table anyway and this returns `true` for `stalled` all the time.